JetMETCorrections/Type1MET/plugins/MultShiftMETcorrDBInputProducer.cc

```diff
@@ -0,0 +1,175 @@
+  mIsData         = cfg.getParameter< bool >("isData");
+
+  pflow_ = consumes<edm::View<reco::Candidate> >(cfg.getParameter< edm::InputTag >("srcPFlow") );
+  vertices_ = consumes<edm::View<reco::Vertex> >( cfg.getParameter< edm::InputTag >("vertexCollection") );
+
+  etaMin_.clear(); 
+  etaMax_.clear();
+
+  produces<CorrMETData>();
+
+
+}
+
+MultShiftMETcorrDBInputProducer::~MultShiftMETcorrDBInputProducer()
+{
+}
+
+void MultShiftMETcorrDBInputProducer::produce(edm::Event& evt, const edm::EventSetup& es)
+{
+  // Get para.s from DB
+  edm::ESHandle<METCorrectorParametersCollection> METCorParamsColl;
+  es.get<METCorrectionsRecord>().get(mPayloadName,METCorParamsColl);
+
+  // get the sections from Collection (pair of section and METCorr.Par class)
+  std::vector<METCorrectorParametersCollection::key_type> keys;
+  // save level to keys for each METParameter in METParameter collection
+  METCorParamsColl->validKeys( keys );
+
+
+  //get primary vertices
+  edm::Handle<edm::View<reco::Vertex> > hpv;
+  evt.getByToken( vertices_, hpv );
+  if(!hpv.isValid()) {
+    edm::LogError("MultShiftMETcorrDBInputProducer::produce") << "could not find vertex collection ";
+  }
+  std::vector<reco::Vertex> goodVertices;
+  for (unsigned i = 0; i < hpv->size(); i++) {
+    if ( (*hpv)[i].ndof() > 4 &&
+       ( fabs((*hpv)[i].z()) <= 24. ) &&
+       ( fabs((*hpv)[i].position().rho()) <= 2.0 ) )
+       goodVertices.push_back((*hpv)[i]);
+  }
+  int ngoodVertices = goodVertices.size();
+
+  edm::Handle<edm::View<reco::Candidate> > particleFlow;
+  evt.getByToken(pflow_, particleFlow);
+
+
+  //MM: loop over all constituent types and sum each correction
+  std::auto_ptr<CorrMETData> metCorr(new CorrMETData());
+  
+  double corx=0.;
+  double cory=0.;
+
+
+  // check DB
+  for ( std::vector<METCorrectorParametersCollection::key_type>::const_iterator ibegin = keys.begin(),
+	  iend = keys.end(), ikey = ibegin; ikey != iend; ++ikey ) {
+    if(mIsData)
+    {
+      if(!METCorParamsColl->isXYshiftData(*ikey) )
+	throw cms::Exception("MultShiftMETcorrDBInputProducer::produce")
+	  << "DB is not for Data. Set proper option: \"corrPfMetXYMultDB.isData\" !!\n";
+    }else{
+      if( METCorParamsColl->isXYshiftData(*ikey) )
+	throw cms::Exception("MultShiftMETcorrDBInputProducer::produce")
+	  << "DB is for Data. Set proper option: \"corrPfMetXYMultDB.isData\" !!\n";
+    }
+  }
+
+  for ( std::vector<METCorrectorParametersCollection::key_type>::const_iterator ibegin = keys.begin(),
+	  iend = keys.end(), ikey = ibegin; ikey != iend; ++ikey ) {
+
+    if( !mIsData){
+
+      if(mSampleType == "MC")
+        if(!METCorParamsColl->isXYshiftMC(*ikey)) continue;
+      if(mSampleType == "DY")
+        if(!METCorParamsColl->isXYshiftDY(*ikey)) continue;
+      if(mSampleType == "TTJets")
+        if(!METCorParamsColl->isXYshiftTTJets(*ikey)) continue;
+      if(mSampleType == "WJets")
+        if(!METCorParamsColl->isXYshiftWJets(*ikey)) continue;
```

Comment:
Please use `if (...) { ... } else if (...) {...} else if` for this series of `if` statements.